Cluster in an undefined state

Hi,

this morning, the cluster went into a funky state. Also, the deployed application is down.
The status result:
```js
{
  "data": {
    "status": "unhealthy",
    "detail": {
      "blobstore": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "controller": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "controller-scheduler": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "controller-worker": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "dashboard": {
        "status": "unhealthy"
      },
      "discoverd": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "flannel": {
        "status": "unhealthy"
      },
      "gitreceive": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "logaggregator": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "postgres": {
        "status": "healthy",
        "version": "v20160218.0"
      },
      "router": {
        "status": "healthy",
        "version": "v20160218.0"
      }
    },
    "version": "v20160218.0"
  }
}
```

```bash
flynn-host ps
```

> 10:05:52.628333 host.go:140: could not list hosts: Get http://127.0.0.1:1111/services/flynn-host/instances: dial tcp 127.0.0.1:1111: connection refused

Collecting the gist gave me in between this:
> INFO[03-03|10:08:08] getting job logs
> EROR[03-03|10:08:38] error getting job logs, falling back to on-disk logs err="could not list hosts: Get http://127.0.0.1:1111/services/flynn-host/instances: dial tcp 127.0.0.1:1111: connection refused"

And the gist of the first node:
https://gist.github.com/a5b524f7d613cd00239e

BTW:
It contained a 0-debug-output.log, but that didn't want to gist due to "Error: "\xC2" on US-ASCII". So I ran it through tr:
```bash
cat 0-debug-output.log | tr -d '\200-\377' > 0-debug-output.clean.log
```

the fix command doesn't work as well:
```
sudo flynn-host fix --min-hosts 3
EROR[03-03|16:13:05] unable to list hosts from discoverd, falling back to peer IP list error="Get http://127.0.0.1:1111/services/flynn-host/instances: dial tcp 127.0.0.1:1111: connection refused"
16:13:05.810540 host.go:140: error connecting to discoverd, use --peer-ips: Get http://127.0.0.1:1111/services/flynn-host/instances: dial tcp 127.0.0.1:1111: connection refused
```

--- a/host/state.go
+++ b/host/state.go
@@ -85,6 +85,17 @@ func (s *State) Restore(backend Backend, buffers host.LogBuffers) (func(), error
 			return err
 		}
 
+		if err := persistentBucket.ForEach(func(k, v []byte) error {
+			for _, job := range s.jobs {
+				if job.Job.ID == string(v) {
+					resurrect = append(resurrect, job.Job)
+				}
+			}
+			return nil
+		}); err != nil {
+			return err
+		}
+
 		// hand opaque blobs back to backend so it can do its restore
 		backendJobsBlobs := make(map[string][]byte)
 		if err := backendJobsBucket.ForEach(func(k, v []byte) error {
@@ -98,18 +109,6 @@ func (s *State) Restore(backend Backend, buffers host.LogBuffers) (func(), error
 			return err
 		}
 
-		s.mtx.Lock()
-		if err := persistentBucket.ForEach(func(k, v []byte) error {
-			for _, job := range s.jobs {
-				if job.Job.ID == string(v) {
-					resurrect = append(resurrect, job.Job)
-				}
-			}
-			return nil
-		}); err != nil {
-			return err
-		}
-		s.mtx.Unlock()
 		return nil
 	}); err != nil && err != io.EOF {
 		return nil, fmt.Errorf("could not restore from host persistence db: %s", err)

